src/sparseml/pytorch/recipe_template/main.py

```diff
@@ -0,0 +1,351 @@
+        is_post_pruning = pruning_algo not in ["constant", ""]
+        quant_builders, quant_variables = _get_quantization_modifier_builders(
+            post_pruning=is_post_pruning, target=target
+        )
+        recipe_variables.update(quant_variables)
+        builder_groups["quantization_modifiers"] = quant_builders
+
+    recipe_builder = RecipeYAMLBuilder(
+        variables=recipe_variables, modifier_groups=builder_groups
+    )
+    recipe = recipe_builder.build_yaml_str()
+
+    if convert_to_md:
+        recipe = f"---\n{recipe}\n---"
+    return recipe
+
+
+def _get_recipe_variables(
+    pruning: bool,
+    quantization: bool,
+    lr_func: str = "linear",
+    mask_type: str = "unstructured",
+    global_sparsity: bool = False,
+) -> Dict[str, Any]:
+    recipe_variables = dict(lr_func=lr_func, init_lr=1.5e-4, final_lr=0)
+
+    if pruning:
+        recipe_variables.update(
+            dict(
+                num_pruning_active_epochs=20,
+                num_pruning_finetuning_epochs=10,
+                pruning_init_sparsity=0.05,
+                pruning_final_sparsity=0.9,
+                pruning_update_frequency=0.01,
+                mask_type=mask_type,
+                global_sparsity=global_sparsity,
+                _num_pruning_epochs=(
+                    "eval(num_pruning_active_epochs + num_pruning_finetuning_epochs)"
+                ),
+            )
+        )
+
+    if quantization:
+        recipe_variables.update(
+            dict(
+                num_qat_epochs=5,
+                num_qat_finetuning_epochs=2.5,
+                quantization_submodules="null",
+            )
+        )
+    if pruning and quantization:
+        recipe_variables["num_epochs"] = "eval(_num_pruning_epochs + num_qat_epochs)"
+    elif pruning and not quantization:
+        recipe_variables["num_epochs"] = "eval(_num_pruning_epochs)"
+    elif quantization and not pruning:
+        recipe_variables["num_epochs"] = "eval(num_qat_epochs)"
+    else:
+        recipe_variables["num_epochs"] = 35
+
+    return recipe_variables
+
+
+def _get_training_modifier_builders() -> List[ModifierYAMLBuilder]:
+    epoch_modifier = ModifierYAMLBuilder(
+        modifier_class=EpochRangeModifier,
+        start_epoch=0.0,
+        end_epoch="eval(num_epochs)",
+    )
+    lr_modifier = ModifierYAMLBuilder(
+        modifier_class=LearningRateFunctionModifier,
+        start_epoch=0.0,
+        end_epoch="eval(num_epochs)",
+        lr_func="eval(lr_func)",
+        init_lr="eval(init_lr)",
+        final_lr="eval(final_lr)",
+    )
+    return [epoch_modifier, lr_modifier]
+
+
+def _get_pruning_modifier_builders(
+    pruning_algo: str,
+    model: Optional[Module] = None,
+    global_sparsity: bool = False,
+) -> Tuple[List[ModifierYAMLBuilder], Dict[str, Any]]:
+    pruning_args = dict(
+        start_epoch=0.0,
+        params="__ALL_PRUNABLE__"
+        if model is None
+        else _get_prunable_params_for_module(model),
```

Comment:
can you make this a local variable above to make scanning easier?

Reply:
Yes!